bluemira/codes/_freecadapi.py

```diff
@@ -667,6 +667,24 @@ def bounding_box(obj) -> Tuple[float, float, float, float, float, float]:
     return box.XMin, box.YMin, box.ZMin, box.XMax, box.YMax, box.ZMax
 
 
+def tesselated_bounding_box(
+    obj, tolerance: float = 1.0
+) -> Tuple[float, float, float, float, float, float]:
+    """
+    Object's bounding box passing through tesselation
+
+    Notes
+    -----
+    Tesselating less trivial geometry objects is the only way to get a more
+    accurate bounding box. Once tesselated, various attributes are modified
+    according to the the tolerance used.
+
+    Lower tolerances will not update the tesselation.
+    """
+    obj.tessellate(tolerance)
```

Comment:
alternatively, we could just expose a `tesselate` method, @ivanmaione ?

Reply:
I am fine with exposing `tasselate`  (or, maybe, `_tasselate` seems more appropriate to me in this case). @je-cook, any thoughts?

Reply:
AS discussed with Ivan, went for _tessellate as a private method on `BluemiraGeo`, left the method for `get_optimal_bounding_box` as a short-hand, so that a user does not need to know about tesselate, but can still access a better bounding box.

Interestingly, as far as I can tell, tessellation does not appear to affect length, area or volume or COG calculations. Just the bounding box, but there may well be situations in which this does occur, or other properties affected, hence the various disclaimers regarding tessellation

Reply:
See my above comment, but I'm a bit tentative about this whole thing